src/Bicep.Core/CoreResources.resx

```diff
@@ -286,4 +286,15 @@
   <data name="MaxNumberResourcesRuleMessageFormat" xml:space="preserve">
     <value>Too many resources. Number of resources is limited to {0}.</value>
   </data>
+  <data name="PreferBarePropertyNamesDeclarationFixTitle" xml:space="preserve">
+    <value>Replace quoted property name with {0}.</value>
+    <comment>{0} is a programmatic language expression</comment>
+  </data>
+  <data name="PreferBarePropertyNamesDereferenceFixTitle" xml:space="preserve">
+    <value>Replace array syntax with {0}.</value>
+    <comment>{0} is a programmatic language expression</comment>
+  </data>
+  <data name="PreferBarePropertyNamesRuleDescription" xml:space="preserve">

```

Comment:
@ucheNkadiCode  Need your magic touch on strings.  Currently:

<img width="772" alt="image" src="https://user-images.githubusercontent.com/6913354/171231810-e77d89e7-4c0a-4ea3-bced-88bc7416aef4.png">
<img width="836" alt="image" src="https://user-images.githubusercontent.com/6913354/171231844-728d752a-89fb-453d-9e93-99a6c90dc9ee.png">

Reply:
I don't see why the second scenario should be a warning. That's a valid way to get the property value. What's the scenario for needing to use obj.myProp instead of obj['myProp']?

Reply:
I think the idea of #5542 was that array access for properties should be more of an escape hatch for when a property name can't be understood as such by the lexer or for when the property name is the result of an expression rather than a single token.

Reply:
My suggestion would be 

"Property {myProp} is a valid identifier and should be declared without quotations and accessed using dot notation"

Reply:
The description also gets used a few places outside of the context of any particular property (like in the [schema file bundled with the VS Code extension](https://github.com/Azure/bicep/blob/main/src/vscode-bicep/schemas/bicepconfig.schema.json)), so I don't believe we can use an interpolated string for the description.

How about "Properties that are valid identifiers should be declared without quotations and accessed using dot notation"? Or if that's too wordy, we could go with something like "Use unquoted property names when possible" (this is more in line with the description for the `prefer-interpolation` rule).

Reply:
Okay thank you for that clarification! I am a fan of either of those options. @StephenWeatherford do you have a preference on the options that @jeskew has presented? 

Either 
 "Properties that are valid identifiers should be declared without quotations and accessed using dot notation"
OR
"Use unquoted property names where possible".

My vote is for the first option and we're good

Reply:
Suggestion minor change:
"Property names that are valid identifiers should be declared without quotations and accessed using dot notation"

I understand there's some ambiguity with property vs property name vs property value.  Personally I think this is close enough and will be understood by everyone.